File: Try_GameEngine5/src/com/example/try_gameengine/action/IRotationController.java
package com.example.try_gameengine.action;


public interface IRotationController {
	public void start(MovementActionInfo info);
	public float getRotation();
	public void setRotation(float rotation);
	public void execute(MovementActionInfo info);
	public void execute(MovementActionInfo info, float t);
	public void reset(MovementActionInfo info);
	public void isInverseAngel();
	public void isCyclePath();
	public void isInversePath();
	public void isWavePath();
	public void isSlopeWavePath();
	public MathUtil getMathUtil();
	public void setMathUtil(MathUtil mathUtil);
	public IRotationController copyNewRotationController();
}

File: Try_GameEngine5/src/com/example/try_gameengine/action/IRotationController.kt
package com.example.try_gameengine.action


interface IRotationController {
    fun start(info: MovementActionInfo?)
    fun getRotation(): Float
    fun setRotation(rotation: Float)
    fun execute(info: MovementActionInfo)
    fun execute(info: MovementActionInfo?, t: Float)
    fun reset(info: MovementActionInfo?)
    fun isInverseAngel()
    fun isCyclePath()
    fun isInversePath()
    fun isWavePath()
    fun isSlopeWavePath()
    fun getMathUtil(): MathUtil?
    fun setMathUtil(mathUtil: MathUtil?)
    fun copyNewRotationController(): IRotationController?
}
